test/nodejs/errorreporting/errorreporting.js

```diff
@@ -0,0 +1,255 @@
+    service_context: {
+      service: SERVICE,
+      version: VERSION
+    },
+    context: {
+      http_request: httpRequestConext,
+      user: USER_NAME,
+      report_location: sourceLocation2
+    }
+  };
+
+  // report the three events in parallel
+  async.eachOf([event1, event2, event3], function(event, index, callback) {
+    api.reportErrorEvent(formattedProjectName, event, function(err, response) {
+      console.log('Reporting error event %d', index + 1);
+      if (err) {
+        console.error(err);
+      }
+      callback(err);
+    });
+  }, function(err) {
+    if (err) {
+      console.error(err);
+    }
+    console.log('createErrorEvents Done');
+    done(err);
+  });
+}
+
+function listErrorInfo(done) {
+  async.waterfall([
+    listErrorGroupStats,
+    listErrorGroupEvents,
+    updateErrorGroup,
+    getErrorGroup
+  ], function(err, result) {
+    console.log('listErrorInfo', result);
+    done(err);
+  });
+}
+
+/**
+ * Integration test for ErrorStatsServiceApi.listGroupStats,
+ *                      ErrorStatsServiceApi.listEvents
+ *
+ */
+function listErrorGroupStats(callback) {
+  var api = clouderrorreportingV1beta1.errorStatsServiceApi();
+  var formattedProjectName = api.projectPath(PROJECT_NAME);
+  var timeRange = {
+    period: clouderrorreportingV1beta1.QueryTimeRange.Period.PERIOD_1_HOUR
+  };
+
+  console.log('Listing all the error groups');
+  var groups = [];
+  api.listGroupStats(formattedProjectName, timeRange).on('data',
+      function(errorGroupStat) {
+        console.log(errorGroupStat.group.name + ' ' +
+            errorGroupStat.group.group_id + ' ' +
+            errorGroupStat.count);
+        groups.push(errorGroupStat);
+      }).on('end', function() {
+        var firstErrorGroupId = groups[0].group.group_id;
+        console.log('firstErrorGroupId = ' + firstErrorGroupId);
+        callback(null, firstErrorGroupId);
+      }).on('error', callback);
+}
+
+function listErrorGroupEvents(errorGroupId, callback) {
+  var api = clouderrorreportingV1beta1.errorStatsServiceApi();
+  var formattedProjectName = api.projectPath(PROJECT_NAME);
+  var timeRange = {
+    period: clouderrorreportingV1beta1.QueryTimeRange.Period.PERIOD_1_HOUR
+  };
+
+  console.log('Listing all the error events for error group ' + errorGroupId);
+  api.listEvents(formattedProjectName, errorGroupId,
+      {timeRange: timeRange}).on('data', function(errorEvent) {
+        console.log(errorEvent.message + ' ' +
+            errorEvent.service_context.service + ' ' +
+            errorEvent.service_context.version);
+        console.log(errorEvent.context.user);
+        console.log(errorEvent.context.http_request.method + ' ' +
+            errorEvent.context.http_request.url);
+        console.log(errorEvent.context.report_location.file_path + ' ' +
+            errorEvent.context.report_location.line_number + ' ' +
+            errorEvent.context.report_location.function_name);
```

Comment:
It is still a bit unclear how beneficial you format the result.

Indeed Node's default `console.log` does some level of pretty-printing, but it does not print greatly on some deeper elements, but isn't it sufficient to use a pretty-printing JSON format?
```javascript
console.log(JSON.stringify(errorEvent, null, 4));
```
might be good enough (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)